### ISSUE TITLE
Does not update after dispatch

### ISSUE THREAD

**Issue by alexdeoliveira** [spiritree/vue-orgchart#10]
Hi, first I want to congratulate you on the package.

I am a beginner in Vuejs and I am not able to display information that is in the "treeArea" variable, which is brought after the dispatch.

![img1](https://user-images.githubusercontent.com/34426/35822536-c32a1804-0a93-11e8-9176-3dc21c723da6.PNG)

AreasTree.vue
```
<template>
  <div>
    <vo-edit :data="treeArea"></vo-edit>
  </div>
</template>

<script>
import { VoEdit } from 'vue-orgchart'
export default {
  data () {
    return {
      treeArea: {
        name: '',
        children: []
      }
    }
  },
  created () {
    this.$store.dispatch('getAreas', this.$route.params.id)
  },
  computed: {
    areaList () {
      return this.$store.state.area.areaList
    }
  },
  watch: {
    areaList (response) {
      this.treeArea = response
    }
  },
  components: {
    VoEdit
  }
}
</script>
```
Can you help me?

**Comment by spiritree** [comment]
Do you mean that you wanna update or re-render the chart if new data is passed?

**Comment by alexdeoliveira** [comment]
No need to update. I just want to display the data that is in the "this. $
Store.state.area.areaList" state variable. But because the dispatch runs
only on the created method, it does not appear to be displaying. Can you
give me an idea how to solve this?

*Aléx de Oliveira*

Desenvolvedor Web Full Stack
skype alex.oliveira.web - (55) 9 8115 8261


2018-02-06 8:10 GMT-02:00 spiritree <notifications@github.com>:

**Comment by spiritree** [comment]
```
<template>
  <div>
    <vo-edit :data="areaList"></vo-edit>
  </div>
</template>

<script>
import { VoEdit } from 'vue-orgchart'
export default {
  data () {
    return {
    }
  },
  created () {
    this.$store.dispatch('getAreas', this.$route.params.id)
  },
  computed: {
    areaList () {
      return this.$store.state.area.areaList
    }
  },
  components: {
    VoEdit
  }
}
</script>
```
Try this?

**Comment by alexdeoliveira** [comment]
Hello @spiritree

I tested your tip, but it still did not show.

See in this video what happens: http://recordit.co/1ACrrEcU8W

Note that after I save a change without updating the browser is displayed.

**Comment by valero90** [comment]
Maybe this would help, I did this cause I had to call ajax via Axios, create a tree from an array and had the same problem:
```
  methods: {
    mountOrgchart () {
      this.$children.forEach((item) => {
        this.orgChartObj = item
        item.orgchart !== undefined ? this.orgchart = item.orgchart : null
      })
    },
```
And then once we capture the created event:
```
this.mountOrgchart()
this.orgChartObj._props.data = this.chartData
this.orgChartObj.initOrgChart()
```

**Comment by alexdeoliveira** [comment]
Hello @valero90 

Thank you so much for your answer. Like I said up here, I'm a beginner. How do I adapt your tip in the code below?

```
<template>
  <div>
    <vo-edit :data="areaList"></vo-edit>
  </div>
</template>

<script>
import { VoEdit } from 'vue-orgchart'
export default {
  data () {
    return {
    }
  },
  created () {
    this.$store.dispatch('getAreas', this.$route.params.id)
  },
  computed: {
    areaList () {
      return this.$store.state.area.areaList
    }
  },
  components: {
    VoEdit
  }
}
</script>
```

Sorry for my ignorance

**Comment by valero90** [comment]
Hi, I think this should work ... I wrote it just here and haven't tested, if you have any problem just tell me 😄 
```
<template>
  <div>
    <vo-edit :data="areaList"></vo-edit>
  </div>
</template>

<script>
import { VoEdit } from 'vue-orgchart'
export default {
  data () {
    return {
      orgChartObj: {}
    }
  },
  created () {
    this.$store.dispatch('getAreas', this.$route.params.id)
    this.mountOrgchart()
    this.orgChartObj._props.data = this.areaList()
    this.orgChartObj.initOrgChart()
  },
  methods: {
    mountOrgchart () {
      this.$children.forEach((item) => {
        this.orgChartObj = item
        item.orgchart !== undefined ? this.orgchart = item.orgchart : null
    })
  },
  computed: {
    areaList () {
      return this.$store.state.area.areaList
    }
  },
  components: {
    VoEdit
  }
}
</script>
```

**Comment by spiritree** [comment]
@valero90
I have tested and it worked.Thanks for your contribution!

**Comment by alexdeoliveira** [comment]
Thank you for your response. But it still was not working.

I was able to solve it in a different way, adding the "then" method after the dispatch and creating a "loading" variable. With this, I display the component only when it returns a "Promise" of the dispatch and the "loading" variable is false.

See the code:
```
<template>
  <div>
    <vo-edit v-if="areaList && !loading" :data="areaList"></vo-edit>
  </div>
</template>
<script>
import { VoEdit } from 'vue-orgchart'
export default {
  data () {
    return {
      loading: true
    }
  },
  created () {
    this.$store.dispatch('getAreas').then(() => {
      this.loading = false
    })
  },
  computed: {
    areaList () {
      return this.$store.state.area.areaList
    }
  },
  components: {
    VoEdit
  }
}
</script>
```

Thank you very much for your help. Maybe this idea of mine is useful to others.

Sorry for my bad english

**Comment by spiritree** [comment]
@alexdeoliveira @valero90 
fixed in version 1.1.2
